arangod/Aql/ExecutionPlan.cpp

```diff
@@ -830,7 +830,10 @@ ExecutionNode* ExecutionPlan::registerNode(ExecutionNode* node) {
   TRI_ASSERT(_ids.find(node->id()) == _ids.end());
 
   try {
-    _ids.emplace(node->id(), node);
+   auto [it, emplaced] = _ids.try_emplace(node->id(), node);
+   if(!emplaced) {
+     throw;
```

Comment:
`throw` what? Is there an active exception here?

Reply:
Ooh - I should have checked the spec. It behaves different than I expected and would result in a terminate. My reasoning was that `...` catches everything and I do not need to create an object that will be discarded anyway.